## 1.0 IMPORT LIBRARIES

In [22]:
import pandas as pd
import numpy as np
import seaborn as sns
from sklearn import metrics
import matplotlib.pyplot as plt
from sklearn.model_selection import train_test_split
from sklearn.tree import DecisionTreeClassifier
from sklearn.metrics import confusion_matrix, accuracy_score, precision_score, recall_score ,f1_score,classification_report, make_scorer
from sklearn.preprocessing import OrdinalEncoder
from sklearn.preprocessing import LabelEncoder
from sklearn.preprocessing import OneHotEncoder
from sklearn.preprocessing import StandardScaler
from sklearn.model_selection import train_test_split
from sklearn.neighbors import KNeighborsClassifier
from sklearn.tree import plot_tree
from sklearn.tree import export_text
from sklearn.utils import resample
from sklearn import preprocessing
from xgboost import XGBClassifier
from sklearn.model_selection import train_test_split, cross_val_score, GridSearchCV, RandomizedSearchCV
from imblearn.over_sampling import RandomOverSampler
from sklearn.ensemble import RandomForestClassifier, GradientBoostingClassifier, AdaBoostClassifier
from sklearn.linear_model import LogisticRegression

# 2.0 LOAD THE DATA

In [2]:
X_train = pd.read_csv('FinancialData_train_X.csv')
X_test = pd.read_csv('FinancialData_test_X.csv')
y_train = pd.read_csv('FinancialData_train_y.csv')
y_test = pd.read_csv('FinancialData_test_y.csv')

In [3]:
X_train.head(5)

ROA(C) before interest and depreciation before interest  \
0                                          -7.064528          
1                                           0.398895          
2                                           1.066762          
3                                          -0.502727          
4                                          -0.152096          

    Operating Gross Margin   Operating Profit Rate  \
0                -0.382990              -24.732582   
1                 0.172025                0.436047   
2                 1.628940                0.436047   
3                -0.313613               -0.531977   
4                -0.729875               -0.531977   

    Pre-tax net Interest Rate   Operating Expense Rate  \
0                   -9.954255                -0.623336   
1                    0.039083                -0.623336   
2                    0.039083                -0.623336   
3                    0.039083                -0.623336   
4                    0.039083                 1.155986   

    Research and development expense rate   Cash flow rate  \
0                               -0.750627        -3.421733   
1                                1.316196         0.185156   
2                               -0.750627         0.735359   
3                               -0.750627        -0.487315   
4                                0.516259        -0.487315   

    Interest-bearing debt interest rate   Tax rate (A)  \
0                             -0.146024      -0.830060   
1                             -0.146024       0.985223   
2                             -0.146024      -0.408400   
3                             -0.146024       3.265046   
4                             -0.146024      -0.830060   

    Net Value Per Share (B)  ...   Current Liability to Current Assets  \
0                 -1.679873  ...                             -0.303571   
1                 -0.089657  ...                              0.236438   
2                  0.137517  ...                             -0.573576   
3                 -0.998352  ...                              3.814000   
4                  0.267331  ...                             -0.168569   

    Liability-Assets Flag   Total assets to GNP price   No-credit Interval  \
0               -0.032383                    4.692638            -0.030519   
1               -0.032383                   -0.053114            -0.030519   
2               -0.032383                   -0.053114            -0.030519   
3               -0.032383                   -0.053114            -0.030519   
4               -0.032383                   -0.053114             0.040261   

    Net Income to Stockholder's Equity   Liability to Equity  \
0                            -1.809277             -0.120046   
1                             0.070906             -0.185017   
2                             0.205204             -0.249988   
3                             0.003756             -0.120046   
4                             0.003756              0.009896   

    Degree of Financial Leverage (DFL)  \
0                            -0.052065   
1                            -0.052065   
2                            -0.052065   
3                             0.055934   
4                             0.001935   

    Interest Coverage Ratio (Interest expense to EBIT)   Net Income Flag  \
0                                           0.013626                 0.0   
1                                           0.013626                 0.0   
2                                           0.013626                 0.0   
3                                           0.157200                 0.0   
4                                           0.085413                 0.0   

    Equity to Liability  
0             -0.201885  
1             -0.096144  
2              0.453711  
3             -0.244182  
4             -0.434516  

[5 rows x 22 columns]

In [4]:
X_train.shape

(4773, 22)

In [5]:
X_train.columns

Index([' ROA(C) before interest and depreciation before interest',
       ' Operating Gross Margin', ' Operating Profit Rate',
       ' Pre-tax net Interest Rate', ' Operating Expense Rate',
       ' Research and development expense rate', ' Cash flow rate',
       ' Interest-bearing debt interest rate', ' Tax rate (A)',
       ' Net Value Per Share (B)', ' CFO to Assets', ' Cash Flow to Equity',
       ' Current Liability to Current Assets', ' Liability-Assets Flag',
       ' Total assets to GNP price', ' No-credit Interval',
       ' Net Income to Stockholder's Equity', ' Liability to Equity',
       ' Degree of Financial Leverage (DFL)',
       ' Interest Coverage Ratio (Interest expense to EBIT)',
       ' Net Income Flag', ' Equity to Liability'],
      dtype='object')

In [6]:
y_train.columns

Index(['Bankrupt'], dtype='object')

# 3.0 Creating dataframe

In [7]:
performance = pd.DataFrame({"model": [], "Accuracy": [], "Precision": [], "Recall": [], "F1": []})

# 4.0 Execution of Models

## Logistic Regression using Random Search CV

In [8]:
score_measure = "recall"
kfolds = 5

parameter_grid = {
    'max_iter':np.arange(500,1000),
    'penalty': ['None','l1','l2','elasticnet'],
    'solver':['saga','liblinear']
}

log_reg = LogisticRegression()
random_search = RandomizedSearchCV(estimator = log_reg, param_distributions=parameter_grid, cv=kfolds, n_iter=500,
                           scoring=score_measure, verbose=1, n_jobs=-1, 
                           return_train_score=True)

_ = random_search.fit(X_train, y_train)

print(f"The best {score_measure} score is {random_search.best_score_}")
print(f"... with parameters: {random_search.best_params_}")\

bestrecall = random_search.best_estimator_

Fitting 5 folds for each of 500 candidates, totalling 2500 fits


/Users/meghanaraju/opt/anaconda3/lib/python3.9/site-packages/sklearn/utils/validation.py:993: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)
/Users/meghanaraju/opt/anaconda3/lib/python3.9/site-packages/sklearn/utils/validation.py:993: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)
/Users/meghanaraju/opt/anaconda3/lib/python3.9/site-packages/sklearn/utils/validation.py:993: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)
/Users/meghanaraju/opt/anaconda3/lib/python3.9/site-packages/sklearn/utils/validation.py:993: DataConversionWarning: A column-vector y was passed when 

The best recall score is 0.14735632183908046
... with parameters: {'solver': 'saga', 'penalty': 'l2', 'max_iter': 508}


/Users/meghanaraju/opt/anaconda3/lib/python3.9/site-packages/sklearn/linear_model/_sag.py:352: ConvergenceWarning: The max_iter was reached which means the coef_ did not converge
  warnings.warn(


In [9]:
c_matrix = confusion_matrix(y_test, random_search.predict(X_test))
TP = c_matrix[1][1]
TN = c_matrix[0][0]
FP = c_matrix[0][1]
FN = c_matrix[1][0]
performance = pd.concat([performance, pd.DataFrame({'model':"Logistic RandomRegression", 
                                                    'Accuracy': [(TP+TN)/(TP+TN+FP+FN)], 
                                                    'Precision': [TP/(TP+FP)], 
                                                    'Recall': [TP/(TP+FN)], 
                                                    'F1': [2*TP/(2*TP+FP+FN)]
                                                     }, index=[0])])

##  Logistic Regression Grid Search CV

In [10]:
score_measure = "recall"
kfolds = 5
max_iter = random_search.best_params_['max_iter']
penalty = random_search.best_params_['penalty']
solver = random_search.best_params_['solver']

parameter_grid = {
    'max_iter': np.arange(max_iter-100,max_iter+100),  
    'penalty': [penalty],
    'solver': [solver]
}

log_reg = LogisticRegression()
grid_search = GridSearchCV(estimator = log_reg, param_grid=parameter_grid, cv=kfolds, 
                           scoring=score_measure, verbose=1, n_jobs=-1,
                           return_train_score=True)

_ = grid_search.fit(X_train, y_train)

print(f"The best {score_measure} score is {grid_search.best_score_}")
print(f"... with parameters: {grid_search.best_params_}")

bestrecall = grid_search.best_estimator_

/Users/meghanaraju/opt/anaconda3/lib/python3.9/site-packages/sklearn/utils/validation.py:993: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)
/Users/meghanaraju/opt/anaconda3/lib/python3.9/site-packages/sklearn/utils/validation.py:993: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)
/Users/meghanaraju/opt/anaconda3/lib/python3.9/site-packages/sklearn/utils/validation.py:993: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)
/Users/meghanaraju/opt/anaconda3/lib/python3.9/site-packages/sklearn/utils/validation.py:993: DataConversionWarning: A column-vector y was passed when 

Fitting 5 folds for each of 200 candidates, totalling 1000 fits


/Users/meghanaraju/opt/anaconda3/lib/python3.9/site-packages/sklearn/linear_model/_sag.py:352: ConvergenceWarning: The max_iter was reached which means the coef_ did not converge
  warnings.warn(
/Users/meghanaraju/opt/anaconda3/lib/python3.9/site-packages/sklearn/linear_model/_sag.py:352: ConvergenceWarning: The max_iter was reached which means the coef_ did not converge
  warnings.warn(
/Users/meghanaraju/opt/anaconda3/lib/python3.9/site-packages/sklearn/utils/validation.py:993: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)
/Users/meghanaraju/opt/anaconda3/lib/python3.9/site-packages/sklearn/linear_model/_sag.py:352: ConvergenceWarning: The max_iter was reached which means the coef_ did not converge
  warnings.warn(
/Users/meghanaraju/opt/anaconda3/lib/python3.9/site-packages/sklearn/linear_model/_sag.py:352: ConvergenceWarning: The max_iter w

The best recall score is 0.14735632183908046
... with parameters: {'max_iter': 408, 'penalty': 'l2', 'solver': 'saga'}


/Users/meghanaraju/opt/anaconda3/lib/python3.9/site-packages/sklearn/linear_model/_sag.py:352: ConvergenceWarning: The max_iter was reached which means the coef_ did not converge
  warnings.warn(


In [11]:
c_matrix = confusion_matrix(y_test, grid_search.predict(X_test))
TP = c_matrix[1][1]
TN = c_matrix[0][0]
FP = c_matrix[0][1]
FN = c_matrix[1][0]
performance = pd.concat([performance, pd.DataFrame({'model':"Logistic GridRegression", 
                                                    'Accuracy': [(TP+TN)/(TP+TN+FP+FN)], 
                                                    'Precision': [TP/(TP+FP)], 
                                                    'Recall': [TP/(TP+FN)], 
                                                    'F1': [2*TP/(2*TP+FP+FN)]
                                                     }, index=[0])])

## SVM Random Search CV using linear, rbf and poly kernal

In [12]:
from sklearn.svm import SVC
score_measure = "recall"
kfolds = 5

parameter_grid = {
    'C': np.arange(1,25),   
    'gamma': ['scale','auto'],
    'kernel':['linear','rbf','poly']
}

svm_model = SVC()
random_search = RandomizedSearchCV(estimator = svm_model, param_distributions=parameter_grid, cv=kfolds, n_iter=500,
                           scoring=score_measure, verbose=1, n_jobs=-1, 
                           return_train_score=True)

_ = random_search.fit(X_train, y_train)

print(f"The best {score_measure} score is {random_search.best_score_}")
print(f"... with parameters: {random_search.best_params_}")


bestrecall = random_search.best_estimator_


/Users/meghanaraju/opt/anaconda3/lib/python3.9/site-packages/sklearn/model_selection/_search.py:292: UserWarning: The total space of parameters 144 is smaller than n_iter=500. Running 144 iterations. For exhaustive searches, use GridSearchCV.
  warnings.warn(
/Users/meghanaraju/opt/anaconda3/lib/python3.9/site-packages/sklearn/utils/validation.py:993: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)
/Users/meghanaraju/opt/anaconda3/lib/python3.9/site-packages/sklearn/utils/validation.py:993: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)
/Users/meghanaraju/opt/anaconda3/lib/python3.9/site-packages/sklearn/utils/validation.py:993: DataConversionWarning: A column-vector y was passed when a 1d array was expected

Fitting 5 folds for each of 144 candidates, totalling 720 fits


/Users/meghanaraju/opt/anaconda3/lib/python3.9/site-packages/sklearn/utils/validation.py:993: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)
/Users/meghanaraju/opt/anaconda3/lib/python3.9/site-packages/sklearn/utils/validation.py:993: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)
/Users/meghanaraju/opt/anaconda3/lib/python3.9/site-packages/sklearn/utils/validation.py:993: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)
/Users/meghanaraju/opt/anaconda3/lib/python3.9/site-packages/sklearn/utils/validation.py:993: DataConversionWarning: A column-vector y was passed when 

The best recall score is 0.13402298850574712
... with parameters: {'kernel': 'poly', 'gamma': 'scale', 'C': 8}


In [13]:
c_matrix = confusion_matrix(y_test, random_search.predict(X_test))
TP = c_matrix[1][1]
TN = c_matrix[0][0]
FP = c_matrix[0][1]
FN = c_matrix[1][0]
performance = pd.concat([performance, pd.DataFrame({'model':"SVM Random", 
                                                    'Accuracy': [(TP+TN)/(TP+TN+FP+FN)], 
                                                    'Precision': [TP/(TP+FP)], 
                                                    'Recall': [TP/(TP+FN)], 
                                                    'F1': [2*TP/(2*TP+FP+FN)]
                                                     }, index=[0])])

## SVM Grid Search CV using linear,rbf and poly kernal

In [14]:
score_measure = "recall"
kfolds = 5

C = random_search.best_params_['C']
gamma = random_search.best_params_['gamma']
kernel = random_search.best_params_['kernel']

parameter_grid = {
    'C': np.arange(C-2,C+2),  
    'gamma': [gamma],
    'kernel': [kernel]
    
}

svm_model = SVC()
grid_search = GridSearchCV(estimator = svm_model, param_grid=parameter_grid, cv=kfolds, 
                           scoring=score_measure, verbose=1, n_jobs=-1,
                           return_train_score=True)

_ = grid_search.fit(X_train, y_train)

print(f"The best {score_measure} score is {grid_search.best_score_}")
print(f"... with parameters: {grid_search.best_params_}")

bestrecall = grid_search.best_estimator_

Fitting 5 folds for each of 4 candidates, totalling 20 fits


/Users/meghanaraju/opt/anaconda3/lib/python3.9/site-packages/sklearn/utils/validation.py:993: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)
/Users/meghanaraju/opt/anaconda3/lib/python3.9/site-packages/sklearn/utils/validation.py:993: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)
/Users/meghanaraju/opt/anaconda3/lib/python3.9/site-packages/sklearn/utils/validation.py:993: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)
/Users/meghanaraju/opt/anaconda3/lib/python3.9/site-packages/sklearn/utils/validation.py:993: DataConversionWarning: A column-vector y was passed when 

The best recall score is 0.13402298850574712
... with parameters: {'C': 8, 'gamma': 'scale', 'kernel': 'poly'}


In [15]:
c_matrix = confusion_matrix(y_test, grid_search.predict(X_test))
TP = c_matrix[1][1]
TN = c_matrix[0][0]
FP = c_matrix[0][1]
FN = c_matrix[1][0]
performance = pd.concat([performance, pd.DataFrame({'model':"SVM Grid", 
                                                    'Accuracy': [(TP+TN)/(TP+TN+FP+FN)], 
                                                    'Precision': [TP/(TP+FP)], 
                                                    'Recall': [TP/(TP+FN)], 
                                                    'F1': [2*TP/(2*TP+FP+FN)]
                                                     }, index=[0])])

## Decision tree by using Random search

In [16]:
score_measure = "recall"
kfolds = 5

parameter_grid = {
    'min_samples_split': np.arange(1,200),  
    'min_samples_leaf': np.arange(1,200),
    'min_impurity_decrease': np.arange(0.0001, 0.01, 0.0005),
    'max_leaf_nodes': np.arange(5, 200), 
    'max_depth': np.arange(1,50), 
    'criterion': ['entropy', 'gini'],
}

dtree = DecisionTreeClassifier()
random_search = RandomizedSearchCV(estimator = dtree, param_distributions=parameter_grid, cv=kfolds, n_iter=500,
                           scoring=score_measure, verbose=1, n_jobs=-1,  # n_jobs=-1 will utilize all available CPUs 
                           return_train_score=True)

_ = random_search.fit(X_train, np.ravel(y_train))

print(f"The best {score_measure} score is {random_search.best_score_}")
print(f"... with parameters: {random_search.best_params_}")

bestrecall = random_search.best_estimator_

Fitting 5 folds for each of 500 candidates, totalling 2500 fits
The best recall score is 0.15448275862068966
... with parameters: {'min_samples_split': 39, 'min_samples_leaf': 7, 'min_impurity_decrease': 0.0011, 'max_leaf_nodes': 99, 'max_depth': 46, 'criterion': 'entropy'}


/Users/meghanaraju/opt/anaconda3/lib/python3.9/site-packages/sklearn/model_selection/_validation.py:372: FitFailedWarning: 
15 fits failed out of a total of 2500.
The score on these train-test partitions for these parameters will be set to nan.
If these failures are not expected, you can try to debug them by setting error_score='raise'.

Below are more details about the failures:
--------------------------------------------------------------------------------
15 fits failed with the following error:
Traceback (most recent call last):
  File "/Users/meghanaraju/opt/anaconda3/lib/python3.9/site-packages/sklearn/model_selection/_validation.py", line 680, in _fit_and_score
    estimator.fit(X_train, y_train, **fit_params)
  File "/Users/meghanaraju/opt/anaconda3/lib/python3.9/site-packages/sklearn/tree/_classes.py", line 937, in fit
    super().fit(
  File "/Users/meghanaraju/opt/anaconda3/lib/python3.9/site-packages/sklearn/tree/_classes.py", line 250, in fit
    raise ValueError(
ValueEr

In [17]:
c_matrix = confusion_matrix(y_test, random_search.predict(X_test))
TP = c_matrix[1][1]
TN = c_matrix[0][0]
FP = c_matrix[0][1]
FN = c_matrix[1][0]
performance = pd.concat([performance, pd.DataFrame({'model':"Dtree_random", 
                                                    'Accuracy': [(TP+TN)/(TP+TN+FP+FN)], 
                                                    'Precision': [TP/(TP+FP)], 
                                                    'Recall': [TP/(TP+FN)], 
                                                    'F1': [2*TP/(2*TP+FP+FN)]
                                                     }, index=[0])])

## Decision tree by using Grid search

In [18]:
score_measure = "recall"
kfolds = 5

param_grid = {
    'min_samples_split': np.arange(30,36),  
    'min_samples_leaf': np.arange(6,12),
    'min_impurity_decrease': np.arange(0.0048, 0.0054, 0.0001),
    'max_leaf_nodes': np.arange(162,168), 
    'max_depth': np.arange(15,21), 
    'criterion': ['entropy'],
}

dtree = DecisionTreeClassifier()
grid_search = GridSearchCV(estimator = dtree, param_grid=param_grid, cv=kfolds, 
                           scoring=score_measure, verbose=1, n_jobs=-1,  # n_jobs=-1 will utilize all available CPUs 
                           return_train_score=True)

_ = grid_search.fit(X_train, np.ravel(y_train))

print(f"The best {score_measure} score is {grid_search.best_score_}")
print(f"... with parameters: {grid_search.best_params_}")

bestrecall = grid_search.best_estimator_

Fitting 5 folds for each of 9072 candidates, totalling 45360 fits
The best recall score is 0.0
... with parameters: {'criterion': 'entropy', 'max_depth': 15, 'max_leaf_nodes': 162, 'min_impurity_decrease': 0.0048, 'min_samples_leaf': 6, 'min_samples_split': 30}


In [19]:
c_matrix = confusion_matrix(y_test, grid_search.predict(X_test))
TP = c_matrix[1][1]
TN = c_matrix[0][0]
FP = c_matrix[0][1]
FN = c_matrix[1][0]
performance = pd.concat([performance, pd.DataFrame({'model':"Dtree_grid", 
                                                    'Accuracy': [(TP+TN)/(TP+TN+FP+FN)], 
                                                    'Precision': [TP/(TP+FP)], 
                                                    'Recall': [TP/(TP+FN)], 
                                                    'F1': [2*TP/(2*TP+FP+FN)]
                                                     }, index=[0])])

/var/folders/f5/wct_d8p94tn748cf64rswjm00000gn/T/ipykernel_15718/732757500.py:8: RuntimeWarning: invalid value encountered in long_scalars
  'Precision': [TP/(TP+FP)],


In [21]:
performance

model  Accuracy  Precision    Recall        F1
0  Logistic RandomRegression  0.964321   0.428571  0.084507  0.141176
0    Logistic GridRegression  0.964321   0.428571  0.084507  0.141176
0                 SVM Random  0.963832   0.400000  0.084507  0.139535
0                   SVM Grid  0.963832   0.400000  0.084507  0.139535
0               Dtree_random  0.965298   0.500000  0.056338  0.101266
0                 Dtree_grid  0.965298        NaN  0.000000  0.000000

# Analysis

Firstly, the table shows results for six different models: Logistic RandomRegression, Logistic GridRegression, SVM Random, SVM Grid, Dtree random, and Dtree grid. And all Models : accuracy, precision, recall, and F1 score were analyzed. Here, we mainly focus on the 'recall' and 'accuracy metrics. 

Recall is an important metric to consider because it measures the proportion of actual positive cases correctly identified by the model. In other words, it evaluates the model's ability to correctly identify all positive cases. A high recall score indicates that the model is able to identify the large percentage of positive cases, whereas a low recall score indicates that the model is missing a large amount of positive cases.

A high recall score is important in this case because it indicates the model's ability to correctly identify businesses that are likely to fail in the future and therefore require insurance coverage. According to the results, the recall scores for all models range from 0.0845 to 0.0563. This suggests that the models aren't very good at predicting which businesses will fail. Further research is needed to identify the reasons of the low recall scores and to improve the models.

Also, from above metrics we can observe "Accuracy" score is almost similar for "logistic regression models" at 96.43%, "SVM models"  at 96.38% and for "Dtree" it is 96.5% respectively. We can conclude from the problem statement and the results that accuracy is an important metric in predicting bankruptcies in the IT industry. In order to avoid future losses, the insurance company must identify which businesses are likely to fail. This means that the models can accurately predict whether or not a business will fail.


